uamqp/message.py

```diff
@@ -264,6 +272,57 @@ def _can_settle_message(self):
             return False
         return True
 
+    @staticmethod
+    def _get_body_type(body):
+        """
+        Automatically detect the MessageBodyType when no body type information is provided.
+        We categorize object of type list/list of lists into ValueType (not into SequenceType) due to
+        compatibility with old uamqp version.
+        """
+        if isinstance(body, (six.text_type, six.binary_type)) or \
+                (isinstance(body, list) and all([isinstance(b, (six.text_type, six.binary_type)) for b in body])):
+            return constants.MessageBodyType.DataType
+        return constants.MessageBodyType.ValueType
+
+    @staticmethod
+    def _validate_body_and_body_type(body, body_type):
+        if body_type == constants.MessageBodyType.DataType:
+            if not (isinstance(body, (six.text_type, six.binary_type)) or
+                    (isinstance(body, list) and all([isinstance(b, (six.text_type, six.binary_type)) for b in body]))):
```

Comment:
Seems a shame to have to do this if-statement twice for what is currently the dominant code path. Do you think it could have a perf impact for high volumes of messages?